src/main/java/com/example/controller/SimulationController.java

```diff
@@ -351,9 +357,12 @@ private void liveTask() {
     }
 
     private void instantFinishTask() {
-        simulation.allowAnimations(false);
         while (!isFinished.get()) {
             processStep();
+
+            if (paused.get()) {
```

Comment:
Couldn't we get rid of this check if we had designed stop functionality as I suggested above?

Reply:
We do need to have some kind of condition here to stop the loop. Otherwise, it will continue executing. We could also set isFinished flag in close() method. WDYT?